File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.cpp
class Solution {
public:
    vector<int> values;
    int kthSmallest(TreeNode* root, int k) {
        dfs(root);
        // sort(values.begin(), values.end());
        return values[k-1];
    }
    void dfs(TreeNode* node) {
        if (!node) return;
        dfs(node->left);
        values.push_back(node->val);
        dfs(node->right);
    }
};

File: 230-kth-smallest-element-in-a-bst/kth-smallest-element-in-a-bst.py
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.result = -1
        self.count = 0
        
        def dfs(node: Optional[TreeNode]) -> None:
            if not node or self.result != -1:
                return
            
            # Traverse the left subtree
            dfs(node.left)
            
            # Process the current node
            self.count += 1
            if self.count == k:
                self.result = node.val
                return
            
            # Traverse the right subtree
            dfs(node.right)
        
        # Start the DFS from the root
        dfs(root)
        return self.result